test/src/code_field/code_controller_comment_uncomment_test.dart

```diff
@@ -0,0 +1,269 @@
+// ignore_for_file: prefer_const_constructors
+
+import 'package:flutter/material.dart';
+import 'package:flutter_code_editor/flutter_code_editor.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:highlight/languages/java.dart';
+
+void main() {
+  group('Selection is collapsed, language: java', () {
+    test(
+        'WHEN selection is collapsed '
+        'SHOULD comment selected line if it is not a comment '
+        'and uncomment otherwise '
+        'AND select whole lines', () {
+      final examples = [
```

Comment:
More ideas for tests:
- ` // A` to comment
- Empty lines to comment out.
- Empty lines to uncomment.
- Different indentation when commenting out.
- Different indentation when uncommenting.